dex.yaml

```diff
@@ -68,8 +68,49 @@ update:
     tag-filter: v
 
 test:
+  environment:
+    contents:
+      packages:
+        - curl
   pipeline:
-    # AUTOGENERATED
     - runs: |
         dex --help
         dex version
+    - uses: test/daemon-check-output
+      with:
+        start: dex serve /etc/dex/config.docker.yaml
+        setup: |
+          cat > /etc/dex/config.docker.yaml << EOF
+          issuer: http://localhost:5556/dex
+          storage:
+            type: memory
+          web:
+            http: 0.0.0.0:5556
+          staticClients:
+            - id: example-app
+              redirectURIs: ["http://localhost:5555/callback"]
+              name: "Example App"
+              secret: ZXhhbXBsZS1hcHAtc2VjcmV0
+          connectors:
+            - type: mockCallback
+              id: mock
+              name: Example
+          oauth2:
+            skipApprovalScreen: true
+          EOF
+        timeout: 30
+        expected_output: |
+          msg="listening on" server=http address=0.0.0.0:5556
+          msg="keys rotated" next_rotation
+        post: |
+          for i in $(seq 1 30); do
+            if curl -s http://localhost:5556/dex/healthz > /dev/null; then
+              break
+            fi
+            if [ $i -eq 30 ]; then
+              echo "Server failed to start"
+              exit 1
+            fi
+            sleep 1
+          done
+          echo "Dex server is up and running"
```

Comment:
instead of all this we can add `wait-for-it` to test environment and then something like following:
```bash
wait-for-it localhost:5556 --timeout=60
curl -fsSL http://localhost:5556/dex/healthz | grep -i "OK"
```
grep for ok if it's returning ok or something else depending on the output.